### PR TITLE
feat: integrated frontend and backend for create account

### PR DESCRIPTION
**Pull request by marin-isabella**
## Description
- Added more input validation checks to the backend.
- Integrated the frontend and backend create account functionality. 
- Refactored the create account frontend code to have one large formState.  

## Milestones
This PR works towards completing milestone 2: user can create an account.

## Resources
[Link to .toISOString() documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString) 

## Test Plan
- The frontend and backend integration was tested by creating a new user on the frontend, and checking if this user was stored in the database. 
- Edge cases I checked include: user inputs incorrect budget range, user does not select a gender or status button, user leaves any other fields empty.

### CODE REVIEW

File: client/src/pages/CreateAccount/CreateAccount.jsx [hunk 1]
```diff
@@ -4,30 +4,65 @@ import LoggedOutNavBar from '../../components/LoggedOutNavBar/LoggedOutNavBar';
...
+    });
+
     const [step, setStep] = useState(1);
     const [passwordError, setPasswordError] = useState('');
+    const [submitError, setSubmitError] = useState('');
+    const [isSubmitting, setIsSubmitting] = useState(false);
+
+    const createUser = async (userData) => {
+        try {
+            const response = await fetch('http://localhost:3000/api/users/create-account', {
```
Comment (scriptist) [inline client/src/pages/CreateAccount/CreateAccount.jsx:34]: You should probably leave out the `localhost` so you can deploy this somewhere. Just make it `/api/users/create-account`, and the browser will work out the rest.

**Review by scriptist — COMMENTED** [review]
Nice work!
